tests/unit/src/test/scala/tests/testProvider/DiscoverTestSuitesSuite.scala

```diff
@@ -0,0 +1,116 @@
+package tests
+
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.testProvider.TestSuiteDiscoveryResult
+
+import tests.TestDiscoveryWrapper._
+
+class DiscoverTestSuitesSuite extends BaseLspSuite("discoverTestSuites") {
+
+  test("munit-test-suite") {
+    for {
+      _ <- Future.successful(cleanWorkspace())
+      _ <- initialize(s"""|/metals.json
+                          |{
+                          |  "app": {
+                          |    "libraryDependencies" : [ "org.scalameta::munit:0.7.29", "org.scalatest::scalatest:3.2.1" ],
+                          |    "scalaVersion": "2.13.6"
+                          |  }
+                          |}
+                          |/app/src/main/scala/foo/bar/MyTestSuite.scala
+                          |package foo.bar
+                          |abstract class AbstractTestSuite1 extends munit.FunSuite 
+                          |
+                          |class MunitTestSuite extends AbstractTestSuite1 {
+                          |  test("foo") {}
+                          |}
+                          |""".stripMargin)
+      _ <- server.server.compilations.compilationFinished(
```

Comment:
```suggestion
      _ <- server.didSave("app/src/main/scala/foo/bar/MyTestSuite.scala")
```

This should be enough and will trigger compilation.